packages/search/src/features/fhir/fhir-utils.ts

```diff
@@ -113,15 +113,22 @@ export async function addEventLocation(
       body.eventLocationId = location.id
     } else {
       body.declarationJurisdictionIds = []
+      if (location.address?.country) {
+        body.eventCountry = location.address.country
+      }
       if (location.address?.state) {
         body.declarationJurisdictionIds.push(location.address.state)
-      } else if (location.address?.district) {
+      }
+      if (location.address?.district) {
         body.declarationJurisdictionIds.push(location.address.district)
-      } else if (location.address?.line?.[10]) {
+      }
+      if (location.address?.line?.[10]) {
         body.declarationJurisdictionIds.push(location.address.line[10])
-      } else if (location.address?.line?.[11]) {
+      }
+      if (location.address?.line?.[11]) {
         body.declarationJurisdictionIds.push(location.address.line[11])
-      } else if (location.address?.line?.[12]) {
+      }
+      if (location.address?.line?.[12]) {
```

Comment:
Let's add some comments mentioning which locationLevel these refer to

Reply:
Done.